lib/bamboo/api_error.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Bamboo.ApiError do
+  @moduledoc """
+  Error used to represent a problem when sending emails through an external email service API.
+  """
+
+  defexception [:message]
+
+  def raise_api_error(message), do: raise(__MODULE__, message: message)
```

Comment:
Can you add docs for this, alone with an example for setting the `extra_message`? I think that could be helpful if an adapter wants to add a bit more details